Extensions/XplatGenerateReleaseNotes/V2/ReleaseNotesFunctions.ts

```diff
@@ -175,246 +185,262 @@ export function processTemplate(template, workItems: WorkItem[], commits: Change
-                                            if ((wi.fields["System.Tags"] !== undefined) &&
-                                                (wi.fields["System.Tags"].toUpperCase() === wiFilter.tags.join("; ").toUpperCase())) {
-                                                agentApi.logDebug (`${addSpace(modeStack.length + 4)} Tags match, need to check fields`);
-                                                okToAdd = true;
+                      // this a new block to add the stack
+                      // need to get the items to process and place them in a queue
+                      agentApi.logDebug (`${addSpace(modeStack.length + 1)} Starting block ${line}`);
+                      // set the index to jump back to
+                      lastBlockStartIndex = index;
+                      switch (mode) {
+                          case Mode.WI:
+                            // store the block and load the first item
+                            // Need to check if we are in tag mode
+                            var modeArray = [];
+                            if (wiFilter.tags.length > 0 || wiFilter.fields.length > 0) {
+                                widetail = undefined;
+                                var parts;
+                                var okToAdd;
+                                workItems.forEach(wi => {
+                                    agentApi.logDebug (`${addSpace(modeStack.length + 2)} Checking WI ${wi.id} with tags '${wi.fields["System.Tags"]}' against tags '${wiFilter.tags.sort().join("; ")}' and fields '${wiFilter.fields.sort().join("; ")}' using comparison filter '${wiFilter.modifier}'`);
+                                    switch (wiFilter.modifier) {
+                                        case Modifier.All:
+                                            agentApi.logDebug (`${addSpace(modeStack.length + 4)} Using ALL filter`);
+                                            if (wiFilter.tags.length > 0) {
+                                                if ((wi.fields["System.Tags"] !== undefined) &&
+                                                    (wi.fields["System.Tags"].toUpperCase() === wiFilter.tags.join("; ").toUpperCase())) {
+                                                    agentApi.logDebug (`${addSpace(modeStack.length + 4)} Tags match, need to check fields`);
+                                                    okToAdd = true;
+                                                } else {
+                                                    agentApi.logDebug (`${addSpace(modeStack.length + 4)} Tags do not match, no need to check fields`);
+                                                    okToAdd = false;
+                                                }
                                             } else {
-                                                agentApi.logDebug (`${addSpace(modeStack.length + 4)} Tags do not match, no need to check fields`);
```

Comment:
This changes behaviour for the existing code path, empty tags did not match previously now it does

Reply:
I checked this was OK